### PR TITLE
INFRA-5854 - add list_streams to boto_kinesis exec module

### PR DESCRIPTION
**Pull request by tkwilliams**
Straightforward `list_streams()` function for boto_kinesis exec module, nothing fancy.

### CODE REVIEW

File: salt/modules/boto_kinesis.py
Comment (ryan-lane) [inline salt/modules/boto_kinesis.py]: Error handling?

Comment (tkwilliams) [reply salt/modules/boto_kinesis.py]: Oop.  Sorry, I misread things and thought _execute_with_retries() was already handling that.  Just pushed an improved version :)

**Comment by cachedout** [comment]
Waiting on re-review from @ryan-lane

File: salt/modules/boto_kinesis.py
Comment (ryan-lane) [inline salt/modules/boto_kinesis.py]: This doesn't seem to be a good way to do the error handling. It takes an error and makes it look like it's not an error. If the backend gives back an error, we should return an error, rather than an empty list.

Comment (cachedout) [reply salt/modules/boto_kinesis.py]: @tkwilliams ?

Comment (tkwilliams) [reply salt/modules/boto_kinesis.py]: Eh, I'm more of a `log.error('Hey, your shiz is broked') ; return []` kinda guy, but the model you're using in the module seems to be `return {'error': 'Hey, your shiz is broked'}` so I can definitely do that to maintain stylistic consistency in the module :)

Should be up in a few minutes...

Comment (ryan-lane) [reply salt/modules/boto_kinesis.py]: We return an error so that the state module can report the errors back up to the end-user in the state output. it also allows us to clearly determine if something was an error, or we really meant to return an empty list, dict, or a None value.

**Comment by rallytime** [comment]
re-run ubuntu

**Review by ryan-lane — APPROVED** [review]

**Pull request merged**
